test/integration/job/job_test.go

```diff
@@ -1084,6 +1084,82 @@ func TestIndexedJob(t *testing.T) {
 	}
 }
 
+func BenchmarkLargeIndexedJob(b *testing.B) {
+	defer featuregatetesting.SetFeatureGateDuringTest(b, feature.DefaultFeatureGate, features.JobTrackingWithFinalizers, true)()
+	closeFn, restConfig, clientSet, ns := setup(b, "indexed")
+	restConfig.QPS = 100
+	restConfig.Burst = 100
+	defer closeFn()
+	ctx, cancel := startJobControllerAndWaitForCaches(restConfig)
+	defer cancel()
+	backoff := wait.Backoff{
+		Duration: time.Second,
+		Factor:   1.5,
+		Steps:    30,
+		Cap:      5 * time.Minute,
+	}
+	mode := batchv1.IndexedCompletion
```

Comment:
As the algorithm for the Indexed jobs saves one API call (https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/2307-job-tracking-without-lingering-pods#simplified-algorithm-for-indexed-jobs) it might be worth to also test the pessimistic scenario of the regular jobs. Don't expect a significant difference though.

Reply:
I see, it covers the typical use case of jobs with high parallelism which are expected to be indexed.